### PR TITLE
Explicitly link to OpenMP

### LINKED ISSUE #220: dlopen symbol not found issue with M1 wheel
I'm getting the following error when trying to import tabmat using the `tabmat-3.1.3-cp38-cp38-macosx_11_0_arm64.whl` wheel:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/sidharthkapur/test-workspace/tabmat-test/lib/python3.8/site-packages/tabmat/__init__.py", line 1, in <module>
    from .categorical_matrix import CategoricalMatrix
  File "/Users/sidharthkapur/test-workspace/tabmat-test/lib/python3.8/site-packages/tabmat/categorical_matrix.py", line 182, in <module>
    from .sparse_matrix import SparseMatrix
  File "/Users/sidharthkapur/test-workspace/tabmat-test/lib/python3.8/site-packages/tabmat/sparse_matrix.py", line 6, in <module>
    from .ext.sparse import (
ImportError: dlopen(/Users/sidharthkapur/test-workspace/tabmat-test/lib/python3.8/site-packages/tabmat/ext/sparse.cpython-38-darwin.so, 0x0002): symbol not found in flat namespace '_je_aligned_alloc'
```

Steps to reproduce:
```sh
python3.8 -m venv ./tabmat-test
source ./tabmat-test/bin/activate
pip install tabmat==3.1.3
# Collecting tabmat
#   Using cached tabmat-3.1.3-cp38-cp38-macosx_11_0_arm64.whl (590 kB)
# Collecting pandas
#   Downloading pandas-1.5.3-cp38-cp38-macosx_11_0_arm64.whl (10.8 MB)
#      |████████████████████████████████| 10.8 MB 15.3 MB/s
# Collecting numpy
#   Using cached numpy-1.24.1-cp38-cp38-macosx_11_0_arm64.whl (13.8 MB)
# Collecting scipy
#   Using cached scipy-1.10.0-cp38-cp38-macosx_12_0_arm64.whl (28.8 MB)
# Collecting python-dateutil>=2.8.1
#   Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
# Collecting pytz>=2020.1
#   Downloading pytz-2022.7.1-py2.py3-none-any.whl (499 kB)
#      |████████████████████████████████| 499 kB 82.4 MB/s
# Collecting six>=1.5
#   Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
# Installing collected packages: six, pytz, python-dateutil, numpy, scipy, pandas, tabmat
# Successfully installed numpy-1.24.1 pandas-1.5.3 python-dateutil-2.8.2 pytz-2022.7.1 scipy-1.10.0 six-1.16.0 tabmat-3.1.3
python -c 'import tabmat'
```

### PR DESCRIPTION
**Pull request by xhochy**
Fixes #220 

Checklist
* [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Comment by xhochy** [comment]
Manually tested that the produced M1 wheel works.

File: build_tools/prepare_macos_wheel.sh [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 
-if [[ "$CIBW_BUILD" == *-macosx_arm64 ]]; then
+if [[ "${ARCHFLAGS:-}" == *arm64 ]]; then
     CONDA_CHANNEL="conda-forge/osx-arm64"
 else
     CONDA_CHANNEL="conda-forge/osx-64"
 fi
 
-conda create -n build -c $CONDA_CHANNEL jemalloc-local xsimd llvm-openmp
+conda create -y -n build -c $CONDA_CHANNEL jemalloc-local xsimd llvm-openmp
```
Comment (jtilly) [inline build_tools/prepare_macos_wheel.sh:11]:
```suggestion
mamba create -y -n build -c $CONDA_CHANNEL jemalloc-local xsimd llvm-openmp
```
?

File: pyproject.toml [hunk 2]
```diff
@@ -45,7 +45,7 @@ before-build = [
 ]
 
 [tool.cibuildwheel.macos.environment]
-LDFLAGS="-Wl,-rpath,$CONDA/envs/build/lib -L$CONDA/envs/build/lib"
+LDFLAGS="-Wl,-rpath,$CONDA/envs/build/lib -L$CONDA/envs/build/lib -lomp -ljemalloclocal"
```
Comment (jtilly) [inline pyproject.toml:48]: (How) is this related to https://github.com/Quantco/tabmat/blob/fbf7755aa48ada1d72dec5d61b7fd9d9f23fdcc6/setup.py#L62-L86 ?

**Review by jtilly — APPROVED** [review]

File: build_tools/prepare_macos_wheel.sh (hunk 1 above)
Comment (xhochy) [reply build_tools/prepare_macos_wheel.sh:11]: We don't have `mamba` installed in this workflow (yet). I can move to micromamba in a follow-up PR. It probably needs a bit of tweaking.

File: pyproject.toml (hunk 2 above)
Comment (xhochy) [reply pyproject.toml]: With the fix above, it might work without these flags again. Let's try.

**Pull request merged**